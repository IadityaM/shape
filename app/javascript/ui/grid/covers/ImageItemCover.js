import { PropTypes as MobxPropTypes } from 'mobx-react'
import styled from 'styled-components'

export const StyledImageCover = styled.div`
  background-image: url(${props => props.url});
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  width: 100%;
  height: 100%;
`
StyledImageCover.displayName = 'StyledImageCover'

class ImageItemCover extends React.PureComponent {
  render() {
    const { item } = this.props
    const { filestack_file_url } = item
    return (
      <StyledImageCover url={filestack_file_url} />
    )
  }
}

ImageItemCover.propTypes = {
  item: MobxPropTypes.objectOrObservableObject.isRequired,
}

export default ImageItemCover
