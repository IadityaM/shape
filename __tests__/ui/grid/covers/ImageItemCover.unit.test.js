import ImageItemCover from '~/ui/grid/covers/ImageItemCover'

import {
  fakeImageItem,
} from '#/mocks/data'

const props = {
  item: fakeImageItem,
}

let wrapper
describe('ImageItemCover', () => {
  beforeEach(() => {
    wrapper = shallow(
      <ImageItemCover {...props} />
    )
  })

  it('renders the StyledImageItem', () => {
    expect(wrapper.find('StyledImageCover').exists()).toBe(true)
  })

  it('passes the url to StyledImageCover', () => {
    expect(wrapper.find('StyledImageCover').props().url).toEqual(fakeImageItem.filestack_file.url)
  })
})
