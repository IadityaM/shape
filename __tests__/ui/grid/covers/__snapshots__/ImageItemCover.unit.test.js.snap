// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageItemCover renders snapshot 1`] = `
<React.Fragment>
  <Viewer
    changeable={false}
    downloadable={false}
    drag={true}
    images={
      Array [
        Object {
          "alt": "",
          "src": "http://fake.url/img",
        },
      ]
    }
    noImgDetails={true}
    noNavbar={true}
    onClose={[Function]}
    onMaskClick={[Function]}
    rotatable={false}
    scalable={false}
    visible={false}
    zIndex={1000}
    zoomSpeed={0.25}
  />
  <styled.div
    onClick={[Function]}
  >
    <SearchIcon />
  </styled.div>
  <StyledImageCover
    contain={false}
    onClick={[Function]}
    url="http://fake.url/img"
  />
</React.Fragment>
`;
